From f76846ab1960eab48e98d3489f4da9da7545b31a Mon Sep 17 00:00:00 2001
From: turbinebmw <turbine.bmw@gmail.com>
Date: Wed, 23 Sep 2026 13:20:42 -0400
Subject: [PATCH 2/3] libipa: camera_sensor_helper: Add IMX681

Convert the Sony reciprocal analogue-gain code to real gain and report the measured 64-code 10-bit black pedestal.
---
 src/ipa/libipa/camera_sensor_helper.cpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/ipa/libipa/camera_sensor_helper.cpp b/src/ipa/libipa/camera_sensor_helper.cpp
index 9457d62..aabbc17 100644
--- a/src/ipa/libipa/camera_sensor_helper.cpp
+++ b/src/ipa/libipa/camera_sensor_helper.cpp
@@ -654,6 +654,18 @@ public:
 };
 REGISTER_CAMERA_SENSOR_HELPER("imx678", CameraSensorHelperImx678)
 
+class CameraSensorHelperImx681 : public CameraSensorHelper
+{
+public:
+	CameraSensorHelperImx681()
+	{
+		/* The black pedestal is 64 at 10 bits. */
+		blackLevel_ = 4096;
+		gain_ = AnalogueGainLinear{ 0, 1024, -1, 1024 };
+	}
+};
+REGISTER_CAMERA_SENSOR_HELPER("imx681", CameraSensorHelperImx681)
+
 class CameraSensorHelperImx708 : public CameraSensorHelper
 {
 public:
-- 
2.55.0

